File: domain/src/main/java/com/movie/domain/dto/MovieProjection.java
package com.movie.domain.dto;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MovieProjection {
    private Long id;
    private String title;
    private String thumbnail;
    private Integer runningTime;
    private String genre;
} 

File: movie/infra/src/main/kotlin/com/example/movie/persistence/movie/projection/MovieProjection.kt
package com.example.movie.persistence.movie.projection

import com.example.movie.domain.movie.model.Rating
import com.example.movie.persistence.genre.projection.GenreProjection
import com.example.movie.persistence.screening.projection.ScreeningProjection
import java.time.LocalDate

interface MovieProjection {
    val id: Long
    val title: String
    val rating: Rating
    val releaseDate: LocalDate
    val thumbnailUrl: String
    val runningTime: Int
    val genre: GenreProjection
    val screenings: List<ScreeningProjection>
}